### PR TITLE
GSdx-d3d: Add support for Depth Conversion on Direct3D11

### PR DESCRIPTION
**Pull request by lightningterror**
Code is ported from OpenGL.
Note: Depth fmt is still not yet supported but after this has been merged it should be a lot easier to do.

Collaborators: @KrossX , @tadanokojin , @gregory38

### CODE REVIEW

File: plugins/GSdx/Renderers/DX11/GSTextureCache11.cpp [hunk 1]
```diff
@@ -62,32 +81,42 @@ void GSTextureCache11::Read(Target* t, const GSVector4i& r)
...
+					break;
+				case PSM_PSMCT24:
+					m_renderer->m_mem.WritePixel24(m.bits, m.pitch, off, r);
+					break;
+				case PSM_PSMCT16:
+				case PSM_PSMCT16S:
+					m_renderer->m_mem.WritePixel16(m.bits, m.pitch, off, r);
+					break;
+
+				case PSM_PSMZ32:
```
[the remark this answers is not in the document]
Comment (lightningterror) [reply plugins/GSdx/Renderers/DX11/GSTextureCache11.cpp]: I think the same can be done for the other formats 24 and 16.

File: plugins/GSdx/Renderers/DX11/GSTextureCache11.cpp [hunk 2]
```diff
@@ -62,32 +77,35 @@ void GSTextureCache11::Read(Target* t, const GSVector4i& r)
...
+					m_renderer->m_mem.WritePixel24(m.bits, m.pitch, off, r);
+					break;
+				case PSM_PSMCT16:
+				case PSM_PSMCT16S:
+				case PSM_PSMZ16:
+				case PSM_PSMZ16S:
+					m_renderer->m_mem.WritePixel16(m.bits, m.pitch, off, r);
+					break;
+				case PSM_PSMCT32:
+				case PSM_PSMZ32:
```
Comment (gregory38) [inline plugins/GSdx/Renderers/DX11/GSTextureCache11.cpp]: You could have put 32/24/16 in a normal order. but that it is fine.

File: plugins/GSdx/Renderers/DX11/GSDevice11.cpp [hunk 3]
```diff
@@ -221,10 +221,14 @@ bool GSDevice11::Create(const std::shared_ptr<GSWnd> &wnd)
 
 	memset(&dsd, 0, sizeof(dsd));
 
-	dsd.DepthEnable = false;
+	hr = m_dev->CreateDepthStencilState(&dsd, &m_convert.dss);
+
+	dsd.DepthEnable = true;
 	dsd.StencilEnable = false;
```
Comment (gregory38) [inline plugins/GSdx/Renderers/DX11/GSDevice11.cpp]: This line is useless. Stencil will be false due to the memset.

**Comment by gregory38** [comment]
It is fine for me (minus 2 nitpicks). It would be nice if it could be reviewed by others people too

**Comment by lightningterror** [comment]
> It would be nice if it could be reviewed by others people too

You are the chosen one :stuck_out_tongue: 
Anyway I took care of the nitpicks as well, I'll go ahead and merge it since all requests/issues have been taken care of. Maybe turtleli can re check the pr just to make sure but I think it's fine.

Time to let normal users enjoy the new goodies.

**Pull request merged**